ext/rotoscope/rotoscope.c

```diff
@@ -138,36 +140,114 @@ static rs_tracepoint_t extract_full_tracevals(rb_trace_arg_t *trace_arg,
 
 static bool in_fork(Rotoscope *config) { return config->pid != getpid(); }
 
-static bool tracecmp(rs_tracepoint_t *a, rs_tracepoint_t *b) {
-  return (!rb_str_cmp(a->method_name, b->method_name) &&
-          !rb_str_cmp(a->entity, b->entity) &&
-          a->method_level == b->method_level);
+// Knowingly produces false positives, but not false negatives
+static bool fast_tracecmp(VALUE tpval1, VALUE tpval2) {
+  rb_trace_arg_t *t1 = rb_tracearg_from_tracepoint(tpval1);
+  rb_trace_arg_t *t2 = rb_tracearg_from_tracepoint(tpval2);
+  return rb_tracearg_method_id(t1) == rb_tracearg_method_id(t2);
```

Comment:
this is used for fast elimination so we can avoid invoking `extract_full_tracevals` on every return event, which would be required to use `tracecmp`.

Reply:
Couldn't we just compare the `self` and `method_id` values from the `call` and `return` events?  That way we can quickly compare the two events with extracting the strings used for logging.